lib/srv/regular/sshserver.go

```diff
@@ -1700,6 +1722,9 @@ func (s *Server) dispatch(ctx context.Context, ch ssh.Channel, req *ssh.Request,
 			s.Logger.Warn(err)
 			return nil
 		}
+		if err := s.termHandlers.SessionRegistry.TryWriteSudoersFile(serverContext); err != nil {
+			return trace.Wrap(err)
```

Comment:
According to the comments in this "case" section, agent forwarding requests should never fail so we should log an error here and return nil.